server.go

```diff
@@ -409,6 +409,16 @@ func (s *ServerImpl) SetMainByPort(port int) error {
 	return s.reloadHaproxy(port)
 }
 
+func (s *ServerImpl) SetMainById(id string) error {
+	for port, deployId := range s.config.Ports {
+		if deployId == id {
+			return s.reloadHaproxy(port)
```

Comment:
Good point, should have a single clear choke point. Added a issue to track this.